### DISCUSSION TITLE
You need to sign in or sign up before continuing

### DISCUSSION THREAD

**Discussion by maaper** [chatwoot/chatwoot#8112] [Self hosted] [answered]
Can someone help me to understand why I receive this message when accessing to this address?
What can be wrong with my installation?

I have Chatwoot running on docker in my Synology NAS. 

![image](https://github.com/chatwoot/chatwoot/assets/56989894/a6ee7591-cafd-4867-9e3b-de7c371dd7a9)

I think that solving this I will solve my issue when trying to link EVOLUTION API with CHATWOOT.

**Comment by maaper** [comment]
Hi all, no clue about that?

I Haver chat won’t running on docker installation.

**Comment by aapeliv** [comment, marked as answer]
For posterity: I had this same problem. It was caused by nginx dropping the `api_access_token` header because it uses underscores. The fix was to add `underscores_in_headers on;` in the `server` block in nginx config. See the nginx config in the repo:

https://github.com/chatwoot/chatwoot/blob/4ed13787d1d359eb3b70c828afb890b0b58c96c8/deployment/nginx_chatwoot.conf#L28

**Reply by diegooptave** [reply]
Thanks man 🙏

**Reply by hugoblanc** [reply]
You saved my day !!

**Comment by imachequa** [comment]
hi there, 

our installation of chatwoot  (AKS: Azure Kubernetes Service, following the helm indication [here](https://www.chatwoot.com/docs/self-hosted/deployment/helm-charts)) is not working with _underscores_in_headers on;_   parameter in nginx.

This is, according to admin console, Chatwoot 3.16.0

Just for our testing purposes we had added below to the nginx configMap and it had helped us:

```
apiVersion: v1
kind: ConfigMap
metadata:
---
data:
  proxy-set-headers: |
    api_access_token: $http_api-access-token
```

On our call, instead of sending api_acces_token header, we sent api-acces-token which is catched by $http_api-access-token and before sending to chatwoot we replaced it to api_acces_token.

I'm omitting a lot of details, that certainly I won't be able to explain since it was done by our DevOps expert.
Hope it helps to anyone out there.

**Comment by luisfrentzen** [comment]
Hi, sorry to upping this, I am still encountering the error despite the config being set to on by default (I am using the latest Chatwoot pull). I am running a local Chatwoot deployment in a Docker container. Is there any way to fix this, or am I missing something?

**Reply by lunadev55** [reply]
Hey, I'm also running a self-hosted Chatwoot version in a Docker container and facing the same issue, already turned the config on but nothing changes. Did you get to solve this problem ?

**Comment by PedroHenriquePdR** [comment]
Hi!

Adding another workaround for those where `underscores_in_headers on;` still doesn't fix the 401 (see @luisfrentzen and @lunadev55 comments above).

Alternative: Use Devise Token Auth instead of `api_access_token`

1. Sign in via the auth endpoint:

```bash
curl -X POST https://your-chatwoot.com/auth/sign_in \
  -H 'Content-Type: application/json' \
  -d '{"email": "your@email.com", "password": "your_password"}'
```

2. Grab these 5 headers from the **response headers** (not the body):

```
access-token: <value>
token-type: Bearer
client: <value>
expiry: <value>
uid: your@email.com
```

3. Include all 5 in every API call:

```bash
curl https://your-chatwoot.com/api/v1/accounts/{id}/contacts \
  -H 'access-token: <value>' \
  -H 'token-type: Bearer' \
  -H 'client: <value>' \
  -H 'expiry: <value>' \
  -H 'uid: your@email.com'
```

This bypasses the underscore issue entirely since all Devise headers use hyphens. Tokens do expire, so you'll need to re-authenticate periodically.

Tested on self-hosted Chatwoot behind Nginx where both `api_access_token` (header and query parameter) returned 401.

**Comment by RafShayder** [comment]
For posterity: I had this same problem. It was caused by nginx dropping the api_access_token header because it uses underscores. The fix was to add underscores_in_headers on; in the server block in nginx config. See the nginx config in the repo

that's true, this is the solution
